.github/workflows/testpythonpackage.yml

```diff
@@ -0,0 +1,55 @@
+# SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+# SPDX-License-Identifier: CC0-1.0
+
+name: Test Python Package
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - '.github/workflows/ghpages.yml'
+      - '.github/workflows/createDockerContainer.yml'
+      - '**.md'
+
+  pull_request:
+    types: [opened]
+    branches:
+      - main
+    paths-ignore:
+      - 'docs/**'
+      - '.github/workflows/ghpages.yml'
+      - '.github/workflows/createDockerContainer.yml'
+      - '**.md'
+
+jobs:
+  build-package:
+    runs-on: ubuntu-latest
+    container:
+      image: ikarusproject/ikarus-dev:latest
+      options: --memory-swap="20g" --memory="20g" --cpus="2" --user root
+
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
I would not use the checkout library, but also install from pip. This would actually really tests if the Python package works because now. It could also just work due to the folder structure of the repo here.

Reply:
The drawback here is that this test can only be done on main, since only then the package is up-to-date to the branch in question.

Reply:
Maybe you can copy the test to another folder. delete the whole repo and then try to execute the tests. Then the test is more similar to the way i suggested.

Reply:
Yeah i think its better to test the current push code and not on main. 
Ive implemented your idea to delete the checked out files